src/main/java/uk/nhs/hee/tis/trainee/forms/service/EventBroadcastService.java

```diff
@@ -0,0 +1,121 @@
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import java.util.Map;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import software.amazon.awssdk.services.sns.SnsClient;
+import software.amazon.awssdk.services.sns.model.MessageAttributeValue;
+import software.amazon.awssdk.services.sns.model.PublishRequest;
+import software.amazon.awssdk.services.sns.model.SnsException;
+import uk.nhs.hee.tis.trainee.forms.dto.LtftFormDto;
+
+/**
+ * A service for broadcasting LTFT form events to SNS.
+ */
+@Slf4j
+@Service
+public class EventBroadcastService {
+
+  protected static final String MESSAGE_ATTRIBUTE_KEY = "trigger";
+
+  private final SnsClient snsClient;
+
+  EventBroadcastService(SnsClient snsClient) {
+    this.snsClient = snsClient;
+  }
+
+  /**
+   * Publish a LTFT form update to SNS.
+   *
+   * @param formDto          The LTFT form DTO to publish.
+   * @param messageAttribute The message attribute to include in the request, or null if not needed.
+   * @param snsTopic         The SNS topic ARN to publish to.
+   */
+  public void publishLtftFormUpdateEvent(LtftFormDto formDto, String messageAttribute,
+      String snsTopic) {
+
+    if (formDto == null) {
+      log.warn("LTFT form is null, skipping SNS publish.");
+      return;
+    }
+    ObjectMapper objectMapper = new ObjectMapper()
+        .registerModule(new JavaTimeModule())
+        .disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
+    JsonNode eventJson = objectMapper.valueToTree(formDto);
+
+    PublishRequest request = buildSnsRequest(snsTopic, eventJson, messageAttribute, formDto.id());
+
+    if (request != null) {
+      try {
+        snsClient.publish(request);
+        log.info("Broadcast event sent to SNS for LTFT form {} with attribute {}.",
+            formDto.id(), messageAttribute);
+      } catch (SnsException e) {
+        String message = String.format(
+            "Failed to broadcast event to SNS topic '%s' for LTFT form '%s'",
+            snsTopic, formDto.id());
+        log.error(message, e);
+      }
+    }
+  }
+
+  /**
+   * Build an SNS publish request.
+   *
+   * @param snsTopic         The SNS topic ARN to publish to.
+   * @param eventJson        The SNS message contents.
+   * @param messageAttribute The message attribute to include in the request, or null if not needed.
+   * @param id               The event UUID.
+   * @return the built request.
+   */
+  private PublishRequest buildSnsRequest(String snsTopic, JsonNode eventJson,
+      String messageAttribute, UUID id) {
+    if (snsTopic == null || snsTopic.isBlank()) {
+      log.warn("SNS topic ARN is null or blank, skipping SNS publish.");
+      return null;
+    }
+
+    PublishRequest.Builder request = PublishRequest.builder()
+        .message(eventJson.toString())
+        .topicArn(snsTopic);
+
+    if (messageAttribute != null) {
```

Comment:
Given previous experiences, I think we need to set a default value (or always require one) or else the message will get DLQ'ed

Reply:
As it happens, there is always an attribute included, but you're right: it is a gotcha. I've added a default.